src/OfficeService.Business/Commands/Office/CreateOfficeCommand.cs

```diff
@@ -24,43 +27,40 @@ public class CreateOfficeCommand : ICreateOfficeCommand
     private readonly IDbOfficeMapper _mapper;
     private readonly ICreateOfficeRequestValidator _validator;
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IResponseCreator _responseCreator;
 
     public CreateOfficeCommand(
       IAccessValidator accessValidator,
       IOfficeRepository officeRepository,
       IDbOfficeMapper mapper,
       ICreateOfficeRequestValidator validator,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IResponseCreator responseCreator)
     {
       _accessValidator = accessValidator;
       _officeRepository = officeRepository;
       _mapper = mapper;
       _validator = validator;
       _httpContextAccessor = httpContextAccessor;
+      _responseCreator = responseCreator;
     }
 
     public async Task<OperationResultResponse<Guid>> ExecuteAsync(CreateOfficeRequest request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.EditCompany))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
       {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
-
-        return new OperationResultResponse<Guid>
-        {
-          Status = OperationResultStatusType.Failed,
-          Errors = new() { "Not enough rights." }
-        };
+        _responseCreator.CreateFailureResponse<Guid>(
+          HttpStatusCode.Forbidden,
+          new() { "Not enough rights." });
```

Comment:
response creator contains the message for badRequest, Forbidden, no need write it